drivers/soundwire/mipi_disco.c

```diff
@@ -493,6 +530,6 @@ int sdw_slave_read_prop(struct sdw_slave *slave)
 	sdw_slave_read_dpn(slave, prop->sink_dpn_prop, nval,
 			   prop->sink_ports, "sink");
 
-	return 0;
+	return sdw_slave_read_lane_mapping(slave);
```

Comment:
another possibility would be to export this function and have codec drivers call it directly, rather than read all properties just to read one that's required.

Reply:
> another possibility would be to export this function and have codec drivers call it directly, rather than read all properties just to read one that's required.

Updated